src/utils/thread_helper.cpp

```diff
@@ -1356,31 +1349,7 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
         }
         // End of WpanRcp section.
 
-        // Start of CoexMetrics section.
-        {
-            auto               coexMetrics = telemetryData.mutable_coex_metrics();
-            otRadioCoexMetrics otRadioCoexMetrics;
-
-            SuccessOrExit(error = otPlatRadioGetCoexMetrics(mInstance, &otRadioCoexMetrics));
-            coexMetrics->set_count_tx_request(otRadioCoexMetrics.mNumTxRequest);
-            coexMetrics->set_count_tx_grant_immediate(otRadioCoexMetrics.mNumTxGrantImmediate);
-            coexMetrics->set_count_tx_grant_wait(otRadioCoexMetrics.mNumTxGrantWait);
-            coexMetrics->set_count_tx_grant_wait_activated(otRadioCoexMetrics.mNumTxGrantWaitActivated);
-            coexMetrics->set_count_tx_grant_wait_timeout(otRadioCoexMetrics.mNumTxGrantWaitTimeout);
-            coexMetrics->set_count_tx_grant_deactivated_during_request(
-                otRadioCoexMetrics.mNumTxGrantDeactivatedDuringRequest);
-            coexMetrics->set_tx_average_request_to_grant_time_us(otRadioCoexMetrics.mAvgTxRequestToGrantTime);
-            coexMetrics->set_count_rx_request(otRadioCoexMetrics.mNumRxRequest);
-            coexMetrics->set_count_rx_grant_immediate(otRadioCoexMetrics.mNumRxGrantImmediate);
-            coexMetrics->set_count_rx_grant_wait(otRadioCoexMetrics.mNumRxGrantWait);
-            coexMetrics->set_count_rx_grant_wait_activated(otRadioCoexMetrics.mNumRxGrantWaitActivated);
-            coexMetrics->set_count_rx_grant_wait_timeout(otRadioCoexMetrics.mNumRxGrantWaitTimeout);
-            coexMetrics->set_count_rx_grant_deactivated_during_request(
-                otRadioCoexMetrics.mNumRxGrantDeactivatedDuringRequest);
-            coexMetrics->set_count_rx_grant_none(otRadioCoexMetrics.mNumRxGrantNone);
-            coexMetrics->set_rx_average_request_to_grant_time_us(otRadioCoexMetrics.mAvgRxRequestToGrantTime);
-        }
-        // End of CoexMetrics section.
+        // TODO: set CoexMetrics by otPlatRadioGetCoexMetrics.
```

Comment:
Should we instead add build config for these and enclose the related code with `#if <config>` check to allow them to be disabled/removed when needed.

This seems to be better way than deleting existing code.

Reply:
Thanks for the suggestion. I will rethink the strategy for this CL and come back later.